core/src/main/java/org/apache/iceberg/view/ViewMetadata.java

```diff
@@ -375,6 +386,13 @@ public ViewMetadata build() {
       Preconditions.checkArgument(null != location, "Invalid location: null");
       Preconditions.checkArgument(versions.size() > 0, "Invalid view: no versions were added");
 
+      // when associated with a metadata file, metadata must have no changes so that the metadata
+      // matches exactly what is in the metadata file, which does not store changes. metadata
+      // location with changes is inconsistent.
+      Preconditions.checkArgument(
```

Comment:
we don't want to fail in the `check()` method but rather in `build()`. Error messaging is aligned with `TableMetadata`

Reply:
I agree with placing this here, although I would note that there's no way to read changes using `ViewMetadataParser`, so that should always produce metadata with a location but no changes. That means even if we did the check in `ViewMetadata`, we wouldn't expect a problem.